fvm/transactionInvoker.go

```diff
@@ -75,8 +75,27 @@ func (i *TransactionInvoker) Process(
 			proc.Events = make([]flow.Event, 0)
 			proc.ServiceEvents = make([]flow.Event, 0)
 		}
+
 		if mergeError := parentState.MergeState(childState, sth.EnforceInteractionLimits()); mergeError != nil {
-			processErr = fmt.Errorf("transaction invocation failed when merging state: %w", mergeError)
+			if processErr != nil {
+				// An error happened while an error was already being handled!
+				// Don't hide the processErr.
+				// But do take the one that is a Failure if one of them is a Failure.
+				var fatal errors.Failure
+				if errors.Is(mergeError, fatal) && !errors.Is(processErr, fatal) {
```

Comment:
This logic seems to me, we will still merge state even if processError is fatal? Why?

Reply:
Probably fees

Reply:
we have to always merge because of spock